mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -115,11 +120,11 @@ def GENX_BlockDimYOp : GENX_DeviceFunctionOp<"workgroup.dim.y",
 def GENX_BlockDimZOp : GENX_DeviceFunctionOp<"workgroup.dim.z",
                         "_Z12get_local_sizej", 2>;
 def GENX_GridDimXOp : GENX_DeviceFunctionOp<"grid.dim.x",
-                        "_Z12get_global_sizej", 0>;
+                        "__builtin_IB_get_global_size", 0>;
```

Comment:
The OpenCL function *should* work. Once IGC fixes the problem you have exposed we should switch back to use the OpenCL function. Can you please add a comment/TODO to the code as a remainder.